cvat/apps/dataset_manager/cron.py

```diff
@@ -0,0 +1,191 @@
+
+    def raise_if_exception(self) -> None:
+        if isinstance(self._exception, Exception):
+            raise self._exception
+
+
+class CleanupTmpDirThread(BaseCleanupThread):
+    description: ClassVar[str] = "Cleanup common temporary directory"
+
+    @suppress_exceptions
+    def _cleanup(self) -> None:
+        # we do not use locks here when handling objects from tmp directory
+        # because undesired race conditions are not possible here:
+        # 1. A temporary file/directory can be removed while checking access time.
+        #    In that case an exception is expected and is handled by the cron process.
+        # 2. A temporary file/directory can be removed by the cron job only when it is outdated.
+        # 3. Each temporary file/directory has a unique name, so the race condition when one process is creating an object
+        #    and another is removing it - impossible.
+        for child in os.scandir(TmpDirManager.TMP_ROOT):
+            # stop clean up process correctly before rq job timeout is ended
+            if self._stop_event.is_set():
+                return
+
+            try:
+                if (
+                    child.stat().st_atime
+                    + timedelta(days=TmpDirManager.TMP_FILE_OR_DIR_RETENTION_DAYS).total_seconds()
+                    < timezone.now().timestamp()
+                ):
+                    if child.is_dir():
+                        shutil.rmtree(child.path)
+                    else:
+                        os.remove(child.path)
+                    logger.debug(f"The {child.name} was successfully removed")
+                    self._number_of_removed_objects += 1
+            except FileNotFoundError:
+                # file or directory has been removed by another process
+                continue
+            except Exception:
+                log_exception(logger)
+
+
+class CleanupExportCacheThread(BaseCleanupThread):
+    description: ClassVar[str] = "Cleanup export cache"
+
+    @suppress_exceptions
+    def _cleanup(self) -> None:
+        export_cache_dir_path = settings.EXPORT_CACHE_ROOT
+        assert os.path.exists(export_cache_dir_path)
+
+        for child in os.scandir(export_cache_dir_path):
+            # stop clean up process correctly before rq job timeout is ended
+            if self._stop_event.is_set():
+                return
+
+            # export cache directory is expected to contain only files
+            if not child.is_file():
+                logger.warning(f"The {child.name} is not a file, skipping...")
+                continue
+
+            try:
+                if clear_export_cache(child):
+                    self._number_of_removed_objects += 1
+            except CacheFileOrDirPathParseError:
+                logger.warning(f"Cannot parse {child.name}, skipping...")
+                continue
+
+            except Exception:
+                log_exception(logger)
+
+
+def cleanup(thread_class_path: str) -> None:
```

Comment:
`python manage.py runperiodicjob cron_export_cache_cleanup`

`TypeError: cleanup() missing 1 required positional argument: 'thread_class_path'`

Reply:
The `runperiodicjob` command was added this week. This command should support passing args rather than setting the default value for the `cleanup` function.

> AttributeError: 'NoneType' object has no attribute 'timeout'

because this command should be executed only from worker